alertlogic/service.py

```diff
@@ -0,0 +1,24 @@
+import requests
+
+class InvalidEndpointCall(Exception):
+    pass
+
+
+class Service:
+
+    def __init__(self):
+        self._session = None
```

Comment:
as mentioned before i think class service should store some default version e.g. "v1", overridable through overloaded init variable. 
```python
def __init__(self, version = "v1"):
```

Reply:
also it is worth delegate url building to the service class as well see(you can almost copy paste it) https://algithub.pd.alertlogic.net/defender/cds_troubleshooting/blob/master/cds/CIAuth.py#L39

Reply:
also i would:
```
class Service:
    def __init__(self, session = None):
        self.set_session(session)
```

and wherever you call it:
```
s = Service(session)
```
instead of calling set_session